doc/memory_management.md

```diff
@@ -0,0 +1,24 @@
+# Memory Management
+
+The function `AdjustExternalMemory` adjusts the amount of registered external
+memory. Used to give the JavaScript engine an indication of the amount of externally
+allocated memory that is kept alive by JavaScript objects.
+The JavaScript engine uses this to decide when to perform global garbage collections.
+Registering externally allocated memory will trigger global garbage collections
+more often than it would otherwise in an attempt to garbage collect the JavaScript
+objects that keep the externally allocated memory alive.
+
+## AdjustExternalMemory
+
+This function gives the JavaScript engine an indication of the amount of externally
+allocated memory that is kept alive by JavaScript objects.
+
+```cpp
+int64_t AdjustExternalMemory(napi_env env, int64_t change_in_bytes);
```

Comment:
I think the env parameter should probably be a node-addon-api  Env as opposed to a napi_env.